lib/web_ui/dev/safari.dart

```diff
@@ -43,21 +35,22 @@ class Safari extends Browser {
         infoLog: DevNull(),
       );
 
-      // Safari will only open files (not general URLs) via the command-line
-      // API, so we create a dummy file to redirect it to the page we actually
-      // want it to load.
-      final Directory redirectDir = Directory(
-        path.join(environment.webUiDartToolDir.path),
-      );
-      final redirect = path.join(redirectDir.path, 'redirect.html');
-      File(redirect).writeAsStringSync(
-          '<script>location = ' + jsonEncode(url.toString()) + '</script>');
-
-      var process =
-          await Process.start(installation.executable, [redirect] /* args */);
-
-      unawaited(process.exitCode
-          .then((_) => File(redirect).deleteSync(recursive: true)));
+      // In the latest versions of MacOs opening Safari browser with a file brings
+      // a popup which halts the test.
+      // The following list of arguments needs to be provided to the `open` command
+      // to open Safari for a given URL. In summary they provide a new instance
+      // to open, that instance to wait for opening the url until Safari launches,
+      // provide Safari bundles identifier.
+      // The details copied from `man open` on MacOS.
+      // TODO(nurhan): https://github.com/flutter/flutter/issues/50809
+      var process = await Process.start(installation.executable, [
```

Comment:
The `Browser` class expects that the process object returned from this function is the browser process, or at least some proxy of it. However, this would return the `open` process that quits immediately. I have a feeling this will become a thorn in future evolution of the `Browser` class. For example, this will prevent us from being able to implement logic that watches the health of the browser process, such as premature termination ([examples](https://github.com/flutter/flutter/search?q=prematurely&unscoped_q=prematurely)).

Having said that, I don't have an immediate solution to this, other than what we talked about offline, i.e. use WebDriver to control launching of browsers. If I'm reading the w3c spec correctly, Selenium is not necessary to use it (see [WebDriver Protocol](https://w3c.github.io/webdriver/#protocol)). So please, treat this comment as non-blocking. I'll file a separate issue for us to investigate a WebDriver solution.

Reply:
Thanks!